### ISSUE TITLE
How does intero-uses-at work?

### ISSUE THREAD

**Issue by JeffreyBenjaminBrown** [chrisdone/intero#612]
When I hit `M-?` on a symbol, other instances of the symbol are highlighted, and the buffer goes into read-only mode. I suspect there's some cool way to navigate between those highlighted instances, maybe even across files, but I'm not finding it by trial and error, and I'm not finding documentation. Also if there's a better way than reloading the buffer to escape intero-uses-at mode, that would be handy.

**Comment by chrisdone** [comment]
It's a small mode with `n`/`p` to jump between, `RET` to exit, and `r` to rename.

**Comment by JeffreyBenjaminBrown** [comment]
Neat!

Now that I know what it does, I may have found a small bug: the rename action does not rename something where its type is stated, only where it is defined and used.

**Comment by chrisdone** [comment]
That's right, a slight limitation of the backend engine. It doesn't take type signatures into account. However, one day I can get more info from the AST. When the current version was written, I was pretty new to the GHC API. Now, I could probably do a better job of it.

**Comment by rmanne** [comment]
The mode doesn't seem to do anything in evil mode. n and p end up overridden by evil commands. What are they supposed to do? Perhaps I can rebind them but I can't figure out what n and p are supposed to be bound to in this mode in the first place.
